File: String/214.Shortest-Palindrome/214.Shortest-Palindrome.cpp
class Solution {
public:
    string shortestPalindrome(string s) 
    {
        if (s=="") return s;
        string t = s;
        reverse(t.begin(),t.end());
        
        int N = s.size();
        int len = N;
        while (s.substr(0,len)!=t.substr(N-len,len))
            len--;
        
        t = s.substr(len);
        reverse(t.begin(),t.end());
        return t+s;
        
    }
};

File: String/214.Shortest-Palindrome/214.Shortest-Palindrome.py
class Solution:
    def shortestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """        
        r = s[::-1]
        i = len(s)
        while not s[0:i]==r[len(s)-i:]:
            i-=1        
        return r[0:len(s)-i]+s
